xmppserver/src/main/java/org/jivesoftware/openfire/group/AbstractGroupProvider.java

```diff
@@ -274,7 +266,7 @@ public boolean hasSharedGroups() {
                 if (DbConnectionManager.getDatabaseType() == DbConnectionManager.DatabaseType.oracle) {
                     pstmt = con.prepareStatement("SELECT 1 FROM dual " + HAS_SHARED_GROUPS_PARTIAL);
                 } else {
-                    pstmt = con.prepareStatement("SELECT 1 " + HAS_SHARED_GROUPS_PARTIAL);
+                    pstmt = con.prepareStatement("SELECT 1 FROM ofGroupProp " + HAS_SHARED_GROUPS_PARTIAL);
```

Comment:
I've updated this line, as it fails on HSQLDB, and so always returns true in the catch block

Reply:
With this, is it needed to differentiate between 'Oracle' and the other databases?

Reading up on this, it appears that HSQLDB should support `con.prepareStatement("VALUES('1') " + HAS_SHARED_GROUPS_PARTIAL`, but:
- I'm not sure if this actually works
- It probably needs to be specific to HSQLDB (adding another branch to that if-statement
- I'm not sure if this has any benefit over `"SELECT 1 FROM ofGroupProp"`

Reply:
> With this, is it needed to differentiate between 'Oracle' and the other databases?

Yeah, looks like the `dual` is the necessary method for Oracle.

> I'm not sure if this has any benefit over `"SELECT 1 FROM ofGroupProp"`

I'd prefer to leave it as-is. This makes sense to me, and is easy to parse.